spectator/registry.py

```diff
@@ -101,12 +101,29 @@ def start(self, config=None):
             logger.debug("registry started with config: %s", config)
             return RegistryStopper(self)
 
-    def stop(self):
+    def clear_meters_and_start(self):
+        """
+        This is called after a fork in the child process
+        to clear the cloned `_meters` and prevent duplicates
+        (the `_meters` are copied with the process
+        during the forking)
+        """
+        self._meters = {}
+        self.start()
+
+    def stop_without_publish(self):
+        """
+        This is called before a fork to prevent a potential deadlock.
+        It cancels the background timer thread. After the fork, the timer
+        thread is restarted in the main and cloned processes.
+        """
         if self._started:
-            logger.info("stopping registry")
+            logger.debug("stopping log registry")
```

Comment:
This isn't really a "log" registry - I would just replicate the function name, to be clear about what is happening in the log message: "stopping registry without publishing".